### DISCUSSION TITLE
UE4_ROOT is not defined, or points to a non-existant directory, please set this environment variable.

### DISCUSSION THREAD

**Discussion by TylerParzialeCat** [carla-simulator/carla#6471] [Q&A]
When attempting to build Carla/Unreal I cannot get past the "make launch" step . I have Unreal version 4.26 which launches on its own and is in a seperate directory, carla I have edited the BuildCarlaUE4.sh , bashrc, and profile to all include UE4_ROOT=~/Unreal-4.26/UnrealEngine-4.26/ which is the path to my Unreal directory. Using both sudo make launch and make launch I am unable to launch carla and it fails with the error : 

BuildCarlaUE4.sh: ERROR: UE4_ROOT is not defined, or points to a non-existant directory, please set this environment variable.
make: *** [Util/BuildTools/Linux.mk:7: launch] Error 2

or 

ERROR: Couldn't touch header directories: Access to the path "/home/tyler/Unreal-4.26/UnrealEngine-4.26/Engine/Intermediate/Build/Linux/B4D820EA/UE4Editor/Inc/CoreUObject/Timestamp" is denied.
make[1]: *** [Makefile:315: CarlaUE4Editor] Error 6
make[1]: Leaving directory '/home/tyler/carla/Unreal/CarlaUE4'
make: *** [Util/BuildTools/Linux.mk:7: launch] Error 2

and am at a loss for how to fix this and be able to launch carla. Help would be greatly appreciated

**Comment by jcastillo-lazr** [comment]
You may want to set your `UE_ROOT=~/Unreal-4.26/` I had a similar problem. On Windows I set it to: `C:\Users\myUser\Workspace\CARLA_UE4\` it applies the same for Windows and Linux.

```
(.venv) C:\Users\myUser\Workspace\carla_0_9_14>echo %UE4_ROOT%
C:\Users\julio.castillo\Workspace\CARLA_UE4\

(.venv) C:\Users\myUser\Workspace\carla_0_9_14>dir C:\Users\myUser\Workspace\CARLA_UE4\

 Directory of C:\Users\myUser\Workspace\CARLA_UE4

06/07/2023  03:15 PM    <DIR>          .
06/07/2023  03:15 PM    <DIR>          ..
06/07/2023  02:53 PM               575 .editorconfig
06/07/2023  02:53 PM               523 .gitattributes
06/07/2023  02:53 PM             6,560 .gitignore
06/07/2023  03:02 PM                72 .tgitconfig
06/07/2023  03:15 PM    <DIR>          .vs
06/07/2023  02:57 PM               456 cpp.hint
06/08/2023  06:46 AM    <DIR>          Engine
06/07/2023  03:00 PM    <DIR>          FeaturePacks
06/07/2023  02:54 PM               660 GenerateProjectFiles.bat
06/07/2023  02:54 PM               238 GenerateProjectFiles.command
06/07/2023  02:54 PM               766 GenerateProjectFiles.sh
06/07/2023  02:54 PM               196 LICENSE.md
06/07/2023  02:54 PM             9,847 README.md
06/07/2023  02:54 PM    <DIR>          Samples
06/07/2023  02:54 PM             1,318 Setup.bat
06/07/2023  02:54 PM               205 Setup.command
06/07/2023  02:54 PM             1,742 Setup.sh
06/07/2023  03:01 PM    <DIR>          Templates
06/07/2023  03:15 PM           718,637 UE4.sln
06/07/2023  02:54 PM               278 UE4Games.uprojectdirs
              15 File(s)        742,073 bytes
               7 Dir(s)  216,869,883,904 bytes free
```

For Linux have you `source` your `.bashrc` after the change, or reboot or open a new terminal?
